code-examples/smart_alert_setting.py

```diff
@@ -0,0 +1,202 @@
+from __future__ import print_function
+import json
+import requests
+from sys import stderr
+
+CERT_FILE = "./certfile"
+
+SMART_ALERT_SETTING_BODY = {
+    "data": {
+        "attributes": {
+            "name": "example",
+            "unit": "bps",
+            "calculation": "pct95",
+            "view": "network",
+            "filters": [
+                {
+                    "facet": "View",
+                    "operator": "=",
+                    "values": [],
+                    "parameters": {
+                    }
+                },
+                {
+                    "facet": "Protocol",
+                    "operator": "=",
+                    "hidden": True,
+                    "values": [],
+                    "parameters": {
+                    }
+                }
+            ],
+            "thresholds": {
+                "medium": 239
+            },
+            "alert_view": {
+                "calculation": "average",
+                "y_axis_scale": "relative",
+                "tab_view": "Relationships",
+                "unit": "pps",
+                "threshold": 1111,
+                "graph_type": "line",
+                "top_contributors": [
+                    "Profile"
+                ],
+                "view": "peer",
+                "filters": [
+                    {
+                        "facet": "Destination IPv4 Address",
+                        "operator": "=",
+                        "hidden": False,
+                        "values": [
+                            "::"
+                        ],
+                        "parameters": {
+                            "output_mask_length": 24
+                        }
+                    }
+                ]
+            }
+        }
+    }
+}
+
+
+def post_smart_alert_setting(leader, key):
+    """POST a smart alert to given leader with given API key.
+
+    Args:
+        leader(string): box to use to create the smart alert setting
+        key(string): an API token granting permission to POST a smart alert on
+            the given leader.
+
+    Returns:
+        string: id of created smart_alert_setting
+    """
+    # build the URL
+    URL = 'https://{leader}/api/sp/smart_alert_settings/'.format(
+        leader=leader)
+
+    # make the API request
+    api_response = requests.post(
+        URL,
+        data=json.dumps(SMART_ALERT_SETTING_BODY),
+        headers={
+            'X-Arbux-APIToken': key,
+            'Content-Type': 'application/vnd.api+json'
+        },
+        verify=CERT_FILE)
+
+    # check the API response
+    if api_response.status_code != requests.codes.created:
+        print("[POST ERROR] API responded with this error: "
+              "{}\n(url: {})".
+              format(api_response.reason, URL),
+              file=stderr)
+        return []
```

Comment:
if the return type is a string, why is an empty list returned here?

Reply:
because it isn't a string :( It's a list of strings (list of IDs). I didn't update the docstring